src/memoize-one.ts

```diff
@@ -0,0 +1,33 @@
+import areInputsEqual from './are-inputs-equal';
+
+export type EqualityFn = (newArgs: unknown[], lastArgs: unknown[]) => boolean;
+
+export default function memoizeOne<
+  // Need to use 'any' rather than 'unknown' here as it has
+  // The correct Generic narrowing behaviour.
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  ResultFn extends (this: any, ...newArgs: any[]) => ReturnType<ResultFn>
```

Comment:
What is `ReturnType`? Is that a TS keyword?

Reply:
It is a typescript utility: https://www.typescriptlang.org/docs/handbook/utility-types.html#returntypet

Reply:
TIL

Reply:
why not ` => any`?